### ISSUE TITLE
Token endpoint throws NPE / HTTP 500

### ISSUE THREAD

**Issue by michael-ylb** [keycloak/keycloak#14045]
### Describe the bug

POST request with empty body to <realm root>/protocol/openid-connect/token causes an HTTP 500 / NPE

### Version

19.0.1

### Expected behavior

An empty request body (as every possible user input) should never cause an HTTP 500 server error. 

Invalid requests should cause a correct error response (whatever is appropriate).

### Actual behavior

HTTP 500 / NPE is thrown

### How to Reproduce?

1. Create new realm. Create new OIDC client. 

2. Send POST request with empty body to <realm root>/protocol/openid-connect/token

3. Observe HTTP 500 and a NPE in the server log

**Comment by lexcao** [comment]
I think it is not a bug.

I have reproduced from my local, and the error logs look like this,
```
2022-09-01 00:32:14,212 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (vert.x-worker-thread-2) Uncaught server error: java.lang.NullPointerException
	at org.jboss.resteasy.plugins.server.BaseHttpRequest.getFormParameters(BaseHttpRequest.java:53)
	at org.jboss.resteasy.plugins.server.BaseHttpRequest.getDecodedFormParameters(BaseHttpRequest.java:74)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.resteasy.core.ContextParameterInjector$GenericDelegatingProxy.invoke(ContextParameterInjector.java:166)
	at com.sun.proxy.$Proxy115.getDecodedFormParameters(Unknown Source)
	at org.keycloak.protocol.oidc.endpoints.TokenEndpoint.processGrantRequest(TokenEndpoint.java:171)
```

The NPE is from `BaseHttpRequest.getFormParameters`, not Keycloak.

I think it may be related to the content type.
If I request with empty body but with header
```
Content-Type: application/x-www-form-urlencoded
```
it will show the expected error message
```
{
	"error": "invalid_request",
	"error_description": "Missing form parameter: grant_type"
}
```

**Comment by michael-ylb** [comment]
Thanks for narrowing it down to the content-type.
So the bug occurs when the content type is missing. 


(For reference: HTTP status code 4xx indicates client errors that the client has to fix, HTTP status code 5xx indicates server errors, that the server has to fix)

**Comment by SaiUpadhyayula** [comment]
@mposolda Can I work on this issue?

**Comment by mposolda** [comment]
@SaiUpadhyayula Sure, Thanks for looking at this

**Comment by SaiUpadhyayula** [comment]
Update: I am still working on this issue and didn't abandon it :D will send out a PR soon

**Comment by klajdiPAJA** [comment]
Hi @mposolda, I was looking at this issue to start working on it but looks to me that the issue has been resolved from changes in #15374.

**Comment by mposolda** [comment]
@klajdiPAJA Cool, Thanks! I am closing this then.
